lib/spack/docs/containers.rst

```diff
@@ -172,6 +164,27 @@ of environments:
          app: "gromacs"
          mpi: "mpich"
 
+To select a proper base image for both the ``build`` and ``final`` stage the ``images`` attribute
+provides two different modes of operation.
+
+The one shown in the example ``spack.yaml`` above requires the user to specify an ``os``
+and a ``spack`` version from a list of possible ones. Spack will then select one of the
+images in :ref:`containers-supported-os` for the build stage and use the base operating
+system image for the final stage.
+
+Alternatively a user might specify explicitly both images:
+
+.. code-block:: yaml
+
+  container:
+    images:
+      build: spack/ubuntu-bionic:latest
+      final: ubuntu:18.04
+
+In this case there's no consistency check performed by Spack, but users are allowed to prepare
+their base images to satisfy custom use cases that cannot be handled with the general purpose
+Spack images distributed on Docker Hub.
```

Comment:
It would be helpful to add some documentation about what, if any, things MUST be present in the build and final images.  At a minimum, point out that `spack containerize` makes *some* assumptions and refer/link to the relevant Dockerfiles as an existence proof of what is known to work.

Reply:
Eheh, to clarify: this is not the final state of the docs and I completely agree with you on the need to specify better the requirements of the two images. I'm just trying to wait to see where we are going before extending this part with more subsections.

Reply:
👍

Reply:
I added a few bits to the docs. Let me know what you think.